egs/madcat_ar/v1/run_end2end.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+# Copyright 2017    Hossein Hadian
+
+set -e
+stage=0
+nj=70
+
+. ./cmd.sh ## You'll want to change cmd.sh to something that will work on your system.
+           ## This relates to the queue.
+. ./path.sh
+. ./utils/parse_options.sh  # e.g. this parses the above options
+                            # if supplied.
+./local/check_tools.sh
+
+
+if [ $stage -le 0 ]; then
+  for dataset in test train dev; do
+    dataset_file=/home/kduh/proj/scale2018/data/madcat_datasplit/ar-en/madcat.$dataset.raw.lineid
+    local/extract_lines.sh --nj $nj --cmd $cmd --dataset_file $dataset_file \
+                           data/local/lines
+  done
+fi
+
+if [ $stage -le 1 ]; then
+  echo "$0: Preparing data..."
+  local/prepare_data.sh
+fi
+
+mkdir -p data/{train,test,dev}/data
+if [ $stage -le 2 ]; then
+  echo "$0: Obtaining image groups..."
+  image/get_image2num_frames.py data/train  # This will be needed for the next command
+  # The next command creates a "allowed_lengths.txt" file in data/train
+  # which will be used by local/make_features.py to enforce the images to
+  # have allowed lengths. The allowed lengths will be spaced by 10% difference in length.
+  image/get_allowed_lengths.py --frame-subsampling-factor 4 10 data/train
+fi
+
+if [ $stage -le 3 ]; then
+  for dataset in test train dev; do
+    local/extract_features.sh --nj $nj --cmd $cmd --feat-dim 40 data/$dataset
+    steps/compute_cmvn_stats.sh data/$dataset || exit 1;
+  done
+  utils/fix_data_dir.sh data/train
+fi
+
+if [ $stage -le 4 ]; then
+  echo "$0: Preparing dictionary and lang..."
+  local/prepare_dict.sh
+  utils/prepare_lang.sh --num-sil-states 4 --num-nonsil-states 8 --sil-prob 0.9999 \
+                        data/local/dict "<sil>" data/lang/temp data/lang
+fi
+
+if [ $stage -le 5 ]; then
+  echo "$0: Estimating a language model for decoding..."
+  local/train_lm.sh
+  utils/format_lm.sh data/lang data/local/local_lm/data/arpa/3gram_unpruned.arpa.gz \
+                     data/local/dict/lexicon.txt data/lang_test
+fi
+
+
+if [ $stage -le 6 ]; then
+  echo "$0: estimating phone language model for the denominator graph"
```

Comment:
In the latest IAM scripts I have moved this stage to the actual chain script (it makes more sense). Please do the same.